tests/distribute_rewards.rs

```diff
@@ -0,0 +1,195 @@
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountOwner)
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_pool_invalid_data() {
+    let holder_rewards_pool = Pubkey::new_unique();
+    let payer = Keypair::new();
+    let amount = 500_000_000_000;
+
+    let mut context = setup().start_with_context().await;
+    setup_system_account(&mut context, &payer.pubkey(), amount).await;
+
+    // Set up a holder rewards pool account with invalid data.
+    {
+        context.set_account(
+            &holder_rewards_pool,
+            &AccountSharedData::new_data(
+                100_000_000,
+                &vec![5; 165],
+                &paladin_rewards_program::id(),
+            )
+            .unwrap(),
+        );
+    }
+
+    let instruction = distribute_rewards(&payer.pubkey(), &holder_rewards_pool, amount);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountData)
+    );
+}
+
+#[tokio::test]
+async fn success() {
+    let holder_rewards_pool = Pubkey::new_unique();
+    let payer = Keypair::new();
+    let amount = 500_000_000_000;
+
+    let mut context = setup().start_with_context().await;
+    setup_system_account(&mut context, &payer.pubkey(), amount).await;
+    setup_holder_rewards_pool_account(&mut context, &holder_rewards_pool, 0).await;
+
+    // For checks later.
+    let payer_beginning_lamports = context
+        .banks_client
+        .get_account(payer.pubkey())
+        .await
+        .unwrap()
+        .unwrap()
+        .lamports;
+
+    let instruction = distribute_rewards(&payer.pubkey(), &holder_rewards_pool, amount);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &payer],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+
+    // Assert the holder rewards pool's total rewards was updated.
+    let holder_rewards_pool_account = context
+        .banks_client
+        .get_account(holder_rewards_pool)
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(
+        bytemuck::from_bytes::<HolderRewardsPool>(&holder_rewards_pool_account.data),
+        &HolderRewardsPool {
+            total_rewards: amount
+        }
+    );
```

Comment:
nit: This is pedantic, but can you also check the `lamports` field on the pool account, just to be totally safe?